src/action-sheets/index.js

```diff
@@ -94,7 +93,10 @@ type MessageArgs = {
   dispatch: Dispatch,
   startEditMessage: (editMessage: EditMessage) => void,
   setDoNotMarkMessagesAsRead: boolean => void,
-  composeBoxRefCurrent: React$ElementRef<ComposeBox> | null,
+  composeBoxRefCurrent: {
+    +doQuoteAndReply: (message: Message | Outbox) => Promise<void>,
+    ...
+  } | null,
```

Comment:
Hmm, this seems like the code gets less readable than it was.

Looking up at the import statement that gets cut:
```js
import typeof ComposeBox from '../compose/ComposeBox';
```
that's a type-only import. The lint rule isn't supposed to apply to those, as mentioned in `tools/import-cycles`:
```
# Tool to explore import cycles, as seen by Flow.
#
# […]
#
# NB these cycles generally do *not* mean an import cycle in the
# transpiled runtime code.  In fact, we use ESLint to prohibit those,
# with rule import/no-cycle.
```

I suspect it has a bug where it recognizes `import type` but not `import typeof`, and accidentally treats the latter the same as plain `import`.

Reply:
Kind of annoying to handle via `eslint-disable`, though, because it's reported on each import in the cycle and so would need such a line on each one of them, most of which are perfectly real runtime imports.

It looks like the import in MessageList isn't part of a cycle. So, perhaps acquiesce to the buggy rule here in action-sheets/, but leave MessageList clean?

Reply:
Or, in place of `React$ElementRef<typeof ComposeBox>`, we could use the `ImperativeHandle` type in ComposeBox.js? I'll try that in my next revision and you can see what you think.

Reply:
OK, and meanwhile sent https://github.com/import-js/eslint-plugin-import/pull/2608 to the relevant ESLint plugin, so hopefully that gets merged and we get to forget about this constraint in the future.